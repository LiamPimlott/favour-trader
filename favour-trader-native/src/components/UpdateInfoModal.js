import React, {Component} from 'react';
import {StyleSheet, View, Modal, KeyboardAvoidingView, Text} from 'react-native';
import {Card, Button, Input} from 'react-native-elements';

class UpdateInfoModal extends React.Component {
    constructor() {
        super();
        this.state = {
            isOpen: null,
            missingFields: false
        };
    }

    componentDidMount() {
        const {isOpen, userProfile} = this.props;
        this.setState({
            isOpen: isOpen,
            firstName: userProfile.firstName,
            lastName: userProfile.lastName,
            country: userProfile.country,
            state: userProfile.state,
            city: userProfile.city,
            postalCode: userProfile.postalCode,
            about: userProfile.about
        })
    }

    update() {
        if (this.state.firstName !== '' &&
            this.state.lastName !== '' &&
            this.state.country !== '' &&
            this.state.state !== '' &&
            this.state.city !== ''
        ) {
           this.props.updateInfo(
                this.state.firstName,
                this.state.lastName,
                this.state.postalCode,
                this.state.city,
                this.state.state,
                this.state.country,
                this.state.about);
            this.props.toggle();
        } else {
            this.setState({missingFields: true});
        }
    }

    render() {
        return (
            <KeyboardAvoidingView behavior="padding">
            <Modal
                animationType="slide"
                transparent={false}
                visible={this.state.isOpen}
                onRequestClose={() => {
                    this.setState({isOpen: false})
                }}>
                    <Card
                        title={'Update Info'}>
                        <View style={{}}>
                            <Input
                                placeholder='First Name'
                                defaultValue={this.state.firstName}
                                onChangeText={(text) => {this.setState({firstName: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='Last Name'
                                defaultValue={this.state.lastName}
                                onChangeText={(text) => {this.setState({lastName: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='Country'
                                defaultValue={this.state.country}
                                onChangeText={(text) => {this.setState({country: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='State'
                                defaultValue={this.state.state}
                                onChangeText={(text) => {this.setState({state: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='City'
                                defaultValue={this.state.city}
                                onChangeText={(text) => {this.setState({city: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='Postal Code'
                                defaultValue={this.state.postalCode}
                                onChangeText={(text) => {this.setState({postalCode: text})}}
                            />
                        </View>
                        <View style={styles.input}>
                            <Input
                                placeholder='About'
                                defaultValue={this.state.about}
                                onChangeText={(text) => {this.setState({about: text})}}
                            />
                        </View>
                        <Button
                            backgroundColor='#03A9F4'
                            buttonStyle={styles.button}
                            title='update Info'
                            onPress={() => {
                               this.update();
                            }}/>
                        {this.state.missingFields ? (<Text>Missing Required Fields</Text>) : (<View />)}
                    </Card>
            </Modal>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    button: {
        marginTop: 10,
    },
    input: {
        marginTop: 10,
        marginBottom: 15,
    },
});

export default UpdateInfoModal;