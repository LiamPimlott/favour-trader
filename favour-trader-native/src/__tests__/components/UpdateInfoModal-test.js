import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import UpdateInfoModal from '../../components/UpdateInfoModal.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing UpdateInfoModal component', () => {
    it('renders as expected', () => {
        const isOpen= false;
        const updateInfo={bind:jest.fn()};
        const toggle={bind:jest.fn()};
        const userProfile={
            firstName: '',
            lastName: '',
            country: '',
            state: '',
            city: '',
            postalCode: '',
            about: ''
        };

        const wrapper = shallow(
            <UpdateInfoModal isOpen={isOpen} updateInfo={updateInfo} toggle={toggle}
                             userProfile={userProfile}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});