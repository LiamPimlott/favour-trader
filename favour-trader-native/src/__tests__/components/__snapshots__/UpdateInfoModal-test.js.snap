// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UpdateInfoModal component renders as expected 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <Card
      title="Update Info"
    >
      <View
        style={Object {}}
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="First Name"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="Last Name"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="Country"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="State"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="City"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="Postal Code"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Input
          defaultValue=""
          onChangeText={[Function]}
          placeholder="About"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <Button
          TouchableComponent={[Function]}
          buttonStyle={
            Object {
              "marginBottom": 15,
              "marginTop": 15,
              "width": 150,
            }
          }
          clear={false}
          disabled={false}
          iconRight={false}
          loadingProps={
            Object {
              "color": "white",
              "size": "small",
            }
          }
          onPress={[Function]}
          title="update Info"
        />
        <Button
          TouchableComponent={[Function]}
          buttonStyle={
            Object {
              "backgroundColor": "red",
              "marginBottom": 15,
              "marginTop": 15,
              "width": 150,
            }
          }
          clear={false}
          disabled={false}
          iconRight={false}
          loadingProps={
            Object {
              "color": "white",
              "size": "small",
            }
          }
          onPress={[Function]}
          title="Close"
        />
      </View>
      <View />
    </Card>
  </Component>
</KeyboardAvoidingView>
`;
